projects/fusion-e2e/tests/components/alert/alert-component.ts

```diff
@@ -1,45 +1,59 @@
-import {Page} from '@playwright/test';
 import {AlertTestIdModifiers} from '@ironsource/fusion-ui/entities';
 import {getTestId, getTestIdSelector} from '../../global/utils';
-
-export class AlertComponent {
-    readonly page: Page;
-
-    constructor(page: Page) {
-        this.page = page;
+import {BaseComponent} from '../base-component';
+import {BaseElement} from '../../behavior';
+import {ButtonComponent} from '../button/button-component';
+
+export class AlertComponent extends BaseComponent {
+    private wrapperElement: BaseElement;
+    private messageElement: BaseElement;
+    private titleElement: BaseElement;
+    private actionButtonElement: ButtonComponent;
+    private closeButtonElement: ButtonComponent;
+
+    constructor(page, selector: string) {
+        super(page, selector);
+        this.wrapperElement = new BaseElement(page, getTestIdSelector(getTestId(this.selector, AlertTestIdModifiers.WRAPPER)));
+        this.messageElement = new BaseElement(page, getTestIdSelector(getTestId(this.selector, AlertTestIdModifiers.MESSAGE)));
+        this.titleElement = new BaseElement(page, getTestIdSelector(getTestId(this.selector, AlertTestIdModifiers.TITLE)));
+        this.actionButtonElement = new ButtonComponent(
+            page,
+            getTestIdSelector(getTestId(this.selector, AlertTestIdModifiers.ACTION_BUTTON))
+        );
+        this.closeButtonElement = new ButtonComponent(page, getTestIdSelector(getTestId(this.selector, AlertTestIdModifiers.CLOSE_BUTTON)));
     }
 
     async waitForComponent({testId}: {testId: string}) {
         const loadedPageSelector = getTestIdSelector(getTestId(testId, AlertTestIdModifiers.WRAPPER));
-        await this.page.waitForSelector(loadedPageSelector);
+        await this.waitForSelector(loadedPageSelector);
```

Comment:
Why does wait for component is implemented here? sounds like a method that should be under baseComponent, also why does it requires a testId param?